dependency/retrieval/components/release.go

```diff
@@ -0,0 +1,84 @@
+package components
+
+import (
+	"bufio"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+)
+
+type Release struct {
+	SemVer *semver.Version
+	URL    string
+
+	Version string `json:"version"`
+}
+
+type Fetcher struct {
+	scriptURL string
+}
+
+func NewFetcher() Fetcher {
+	return Fetcher{
+		scriptURL: "https://aka.ms/getvsdbgsh",
+	}
+}
+
+func (f Fetcher) WithScriptURL(url string) Fetcher {
+	f.scriptURL = url
+	return f
+}
+
+func (f Fetcher) Get() ([]Release, error) {
+	response, err := http.Get(f.scriptURL)
+	if err != nil {
+		return nil, err
+	}
+	defer response.Body.Close()
+
+	if !(response.StatusCode >= 200 && response.StatusCode < 300) {
+		return nil, fmt.Errorf("received a non 200 status code from %s: status code %d received", f.scriptURL, response.StatusCode)
+	}
+
+	scanner := bufio.NewScanner(response.Body)
+
+	var version string
+	var inFunction, latest bool
+	for scanner.Scan() {
+		if inFunction && latest {
+			version = strings.Split(strings.TrimSpace(scanner.Text()), "=")[1]
+			break
+		}
+
+		if inFunction && strings.Contains(scanner.Text(), "latest)") {
+			latest = true
+			continue
+		}
+
+		if strings.Contains(scanner.Text(), "set_vsdbg_version()") {
+			inFunction = true
+			continue
+		}
+
+	}
+
+	splitVersion := strings.Split(version, ".")
+	if len(splitVersion) != 4 {
```

Comment:
As written, if the `set_vsdbg_version()` function changes in the script, retrieval will fail because `splitVersion=[]string{""}`.  This is a bit opaque. Consider explicitly checking if `inFunction` and/or `latest` remain false after the scanner has searched the entire file and failing accordingly. This will explicitly indicate when the script has changed in a way we didn't expect.

Reply:
I think that I have added some valuable lines. Please give it once over and tell me what you think!

Reply:
👍 That works for me!